Wikipedia/Code/AppDelegate.m

```diff
@@ -67,6 +67,10 @@ - (void)updateDynamicIconShortcutItems {
 #pragma mark - UIApplicationDelegate
 
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    NSString *lastAppVersion = NSUserDefaults.wmf_userDefaults.wmf_lastAppVersion;
+    if (!lastAppVersion) {
```

Comment:
This will only be updated once when `lastAppVersion` is `nil` - it needs an `isEqualToString:` check. You could do `if (appVersion && ![lastAppVersion isEqualToString:appVersion])` and that should cover nil & inequality cases

Reply:
Shouldn't it be updated after I log a new snapshot? If I check if `lastAppVersion == appVersion`  when app launches and set `lastAppVersion` to `appVersion`, I will lose `lastAppVersion` and the check in snapshot log will fail

Reply:
Yeah that makes sense - or explicitly log the snapshot here and remove the check from the funnel